process/block/baseProcess.go

```diff
@@ -2513,6 +2513,12 @@ func (bp *baseProcessor) saveMiniBlocksFromExecutionResults(header data.HeaderHa
 		if err != nil {
 			return err
 		}
+
+		executionResultHeaderHash := baseExecutionResult.GetHeaderHash()
+		// cleanup all intra shard miniblocks
+		bp.dataPool.MiniBlocks().Remove(executionResultHeaderHash)
+		// cleanup all log events
+		bp.dataPool.PostProcessTransactions().Remove(common.PrepareLogEventsKey(executionResultHeaderHash))
```

Comment:
maybe we can extract this into a separate function, and call it from `CommitBlock`, to have it called explicitly
- there is already `cleanupPools` function that might be correlated with this

Reply:
refactored